RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitMultiCollectionProducer.cc

```diff
@@ -0,0 +1,93 @@
+// Author: Felice Pantaleo (CERN), 2025, felice.pantaleo@cern.ch
+//
+// Produce a MultiCollection<HGCRecHitCollection> that bundles the EE, FH
+// and BH rec‑hit branches into one flat, persistent object.  Analysis modules
+// such as RecHitMapProducer can consume the manager and access the hits via the
+// usual MultiSpan API.
+//
+// Configuration example:
+// cms.EDProducer("HGCalRecHitMultiCollectionProducer",
+//                EEInput = cms.InputTag("HGCalRecHit", "HGCEERecHits"),
+//                FHInput = cms.InputTag("HGCalRecHit", "HGCHEFRecHits"),
+//                BHInput = cms.InputTag("HGCalRecHit", "HGCHEBRecHits"))
+
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/Common/interface/MultiCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHit.h"
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+
+class HGCalRecHitMultiCollectionProducer : public edm::global::EDProducer<> {
+public:
+  explicit HGCalRecHitMultiCollectionProducer(edm::ParameterSet const& ps) {
+    std::vector<edm::InputTag> tags = ps.getParameter<std::vector<edm::InputTag>>("hits");
+    for (auto& tag : tags) {
+      if (tag.label().find("HGCalRecHit") != std::string::npos) {
+        hgcal_hits_token_.push_back(consumes<HGCRecHitCollection>(tag));
+      } else {
+        barrel_hits_token_.push_back(consumes<reco::PFRecHitCollection>(tag));
+      }
+    }
+
+    produces<edm::MultiCollection<HGCRecHitCollection>>();
+    produces<edm::MultiCollection<reco::PFRecHitCollection>>();
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<std::vector<edm::InputTag>>("hits",
+                                         {edm::InputTag("HGCalRecHit", "HGCEERecHits"),
+                                          edm::InputTag("HGCalRecHit", "HGCHEFRecHits"),
+                                          edm::InputTag("HGCalRecHit", "HGCHEBRecHits")});
+    descriptions.add("hgcalRecHitMultiCollectionProducer", desc);
+  }
```

Comment:
Isn't this configuration going to segfault when the code tries to access `barrel_hits_token_[0]` or `barrel_hits_token_[1]` ?